modules/clock_info.js

```diff
@@ -37,57 +37,74 @@ exports.load = function() {
   // info used for drawing...
   var hrm = "--";
   var alt = "--";
+  var interval = null;
   // callbacks (needed for easy removal of listeners)
-  function batteryUpdateHandler() { items[0].emit("redraw"); }
-  function stepUpdateHandler() { items[1].emit("redraw"); }
-  function hrmUpdateHandler() { items[2].emit("redraw"); }
+  function batteryUpdateHandler() { bangleItems[0].emit("redraw"); }
+  function stepUpdateHandler() { bangleItems[1].emit("redraw"); }
+  function hrmUpdateHandler() { bangleItems[2].emit("redraw"); }
   function altUpdateHandler() {
     Bangle.getPressure().then(data=>{
       if (!data) return;
       alt = Math.round(data.altitude) + "m";
-      items[3].emit("redraw");
+      bangleItems[3].emit("redraw");
     });
   }
+  function deleteInterval(inter){
+    if(inter){
+      clearInterval(inter);
+    }
+    delete inter;
```

Comment:
This doesn't work I'm afraid. It will just delete the argument `inter` but not the actual field.

So `clearInterval(this.interval); delete this.interval;` is needed - or you could do `deleteInterval(this)` and then change this to `clearInterval(obj.interval);delete obj.interval;` - but it doesn't allow for >1 interval (all intervals would have to just be called `this.interval`)

Reply:
I use this.interval right now. Thanks for this hint!